src/app/shared/component/modal/modal.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, OnDestroy, OnInit, Inject } from '@angular/core';
+import { carDetailsService } from '../../../services/car-details.service';
+import { Subscription } from 'rxjs';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+@Component({
+  selector: 'app-modal',
+  templateUrl: './modal.component.html',
+  styleUrl: './modal.component.css',
+})
+export class ModalComponent implements OnInit, OnDestroy {
```

Comment:
Ten komponent modala nie powinien być jal dla mnie w shared. Jest to specyficzny modal dla konkretnego przypadku.